heuristic/repair_operators/_yth_best_insert.py

```diff
@@ -0,0 +1,41 @@
+import heapq
+from numpy.random import RandomState
+
+from heuristic.classes import Route, Solution
+from heuristic.functions import create_single_customer_route
+
+
+def _yth_best_insert(min_offset: int, current: Solution,
```

Comment:
I'd call this `_random_near_best_greedy_insert` I think. That's much more descriptive, and avoids using 'y', which has no meaning in our code. Do _not_ use 'y', it's completely meaningless.

Then `min_offset` can become `max_offset`, with an explanation of how much worse it could potentially be (at most max_offset distance from the optimal insertion point).